### PR TITLE
Prevent callback being called twice

### PR DESCRIPTION
**Pull request by gmasclet**
See explanation on https://github.com/jsumners/node-activedirectory/issues/92#issuecomment-1658435161

### CODE REVIEW

File: lib/components/search.js [hunk 1]
```diff
@@ -26,6 +26,18 @@ function isReferralAllowed (referralUri) {
   return result
 }
 
+function callOnlyOnce (callback) {
+  var hasBeenCalled = false
+  return function () {
+    if (hasBeenCalled) {
+      log.trace('Searcher.search callback already called. Args=%s', JSON.stringify(arguments))
```
Comment (Diachk) [inline lib/components/search.js:33]: The log variable looks uninitialised, it is not clear to me how this works.

**Review by Diachk — APPROVED** [review]
LGTM except I have one question on how log works.

File: lib/components/search.js (hunk 1 above)
Comment (gmasclet) [reply lib/components/search.js:33]: It's defined at the beginning of this file:

https://github.com/Linkurious/node-activedirectory2/blob/046a83ad445b98a9ded7c4687ebd365e3e766491/lib/components/search.js#L8-L10

And it's initialized by the exported function at the end of the file:

https://github.com/Linkurious/node-activedirectory2/blob/046a83ad445b98a9ded7c4687ebd365e3e766491/lib/components/search.js#L355-L360

Finally, the main file inject the actual value:

https://github.com/Linkurious/node-activedirectory2/blob/046a83ad445b98a9ded7c4687ebd365e3e766491/lib/activedirectory.js#L11

https://github.com/Linkurious/node-activedirectory2/blob/046a83ad445b98a9ded7c4687ebd365e3e766491/lib/activedirectory.js#L258

**Comment by Use4ul** [comment]
Thanks!

**Pull request merged**